internal/pkg/bulk/engine.go

```diff
@@ -115,36 +115,46 @@ func NewBulker(es esapi.Transport, tracer *apm.Tracer, opts ...BulkOpt) *Bulker
 	}
 
 	return &Bulker{
-		opts:                  bopts,
-		es:                    es,
-		ch:                    make(chan *bulkT, bopts.blockQueueSz),
-		blkPool:               sync.Pool{New: poolFunc},
-		apikeyLimit:           semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
-		tracer:                tracer,
-		remoteOutputConfigMap: make(map[string]map[string]interface{}),
-		// remote ES bulkers
-		bulkerMap: make(map[string]Bulk),
+		opts:        bopts,
+		es:          es,
+		ch:          make(chan *bulkT, bopts.blockQueueSz),
+		blkPool:     sync.Pool{New: poolFunc},
+		apikeyLimit: semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
+		tracer:      tracer,
 	}
 }
 
 func (b *Bulker) GetBulker(outputName string) Bulk {
-	return b.bulkerMap[outputName]
+	o, ok := b.bulkerMap.Load(outputName)
+	if !ok {
+		return nil
+	}
+	return o.(Bulk)
```

Comment:
Can this panic?

Reply:
it shouldn't, we are only adding Bulkers